Content.Shared/Emag/Systems/EmagSystem.cs

```diff
@@ -156,12 +156,28 @@ public bool TryUseEmag(EntityUid emag, EntityUid user, EntityUid target, EmagCom
         /// </summary>
         public bool DoEmagEffect(EntityUid user, EntityUid target)
         {
+            // prevent emagging twice
+            if (IsEmagged(target))
+                return false;
+
             var emaggedEvent = new GotEmaggedEvent(user);
             RaiseLocalEvent(target, ref emaggedEvent);
+
+            // add marker component unless the handler wants repeatable emagging
+            if (emaggedEvent.Emag)
+                EntityManager.AddComponent<EmaggedComponent>(target);
```

Comment:
In systems you should use the EntitySystem.Proxy methods instead.

Reply:
wasnt aware of it back when i wrote it, thanks for spotting